psycopg_pool/psycopg_pool/pool.py

```diff
@@ -627,6 +627,23 @@ def _connect(self, timeout: float | None = None) -> CT:
         self._set_connection_expiry_date(conn)
         return conn
 
+    def _resolve_conninfo(self) -> str | Any:
+        """Resolve conninfo (static string, sync callable, or async callable)."""
+        if callable(self.conninfo):
+            return self.conninfo()
+
+        return self.conninfo
+
+    def _resolve_kwargs(self) -> dict[str, Any] | Any:
```

Comment:
I don't expect Any to be needed. But maybe there is a problem in the sync pool because we sort of declare that these callbacks might be async, whereas the reality is that they can only be sync in the sync pool?

If that's the case maybe we should move the type annotations from the base class to the subclasses.

```suggestion
    def _resolve_conninfo(self) -> str:
        """Resolve conninfo (static string, sync callable, or async callable)."""
        if callable(self.conninfo):
            return self.conninfo()

        return self.conninfo

    def _resolve_kwargs(self) -> dict[str, Any]:
```